Dockerfile

```diff
@@ -17,37 +17,48 @@ WORKDIR /build
 # as long as your Package.swift/Package.resolved
 # files do not change.
 COPY ./Package.* ./
-RUN swift package resolve
+RUN swift package resolve --skip-update \
+        $([ -f ./Package.resolved ] && echo "--force-resolved-versions" || true)
 
 # Copy entire repo into container
 COPY . .
 
-# Build everything, with optimizations
-RUN swift build -c release --static-swift-stdlib --product Penny
+# Build everything, with optimizations, with static linking, and using jemalloc
+# N.B.: The static version of jemalloc is incompatible with the static Swift runtime.
+RUN swift build -c release \
+      --target Penny \
+      --static-swift-stdlib \
+      -Xlinker -ljemalloc
```

Comment:
`--target Penny` instead of `--product` because i removed the product as it seemed unnecessary.